seb_openedx/user_banning.py

```diff
@@ -0,0 +1,93 @@
+""" Banned users handling module """
+import abc
+from datetime import datetime
+from django.conf import settings
+from django.utils import six
+from seb_openedx.models import ForbiddenCourseAccess
+
+
+def is_user_banned(username, course_key):
+    """ exposed function to check if user is already banned """
+    return _get_back_end().is_user_banned(username, course_key)
+
+
+def ban_user(username, course_key, banned_by):
+    """ exposed function to check if user is already banned """
+    last_modified_time = datetime.now()
+    _get_back_end().ban_user(username, course_key, last_modified_time, banned_by)
+
+
+def unban_user(username, course_key, banned_by):
+    """ exposed function to check if user is already banned """
+    last_modified_time = datetime.now()
+    _get_back_end().unban_user(username, course_key, last_modified_time, banned_by)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BannedUsersBackend(object):
+    """ Abstract BannedUserBackend class """
+
+    @abc.abstractmethod
+    def is_user_banned(self, username, course_key):
+        """ is_user_banned abstract method """
+        pass
+
+    @abc.abstractmethod
+    def ban_user(self, username, course_key, last_modified_time, banned_by):
+        """ is_user_banned abstract method """
+        pass
+
+    @abc.abstractmethod
+    def unban_user(self, username, course_key, last_modified_time, unbanned_by):
+        """ is_user_banned abstract method """
+        pass
+
+
+class DatabaseBannedUsersBackend(BannedUsersBackend):
+    """ Database backend """
+    def is_user_banned(self, username, course_key):
+        """ is_user_banned implementation """
```

Comment:
This docstring could give more info. It is clearly a placeholder, but now it makes sense to actually write it

Reply:
Ok